lib/mayaUsd/nodes/proxyAccessor.py

```diff
@@ -0,0 +1,286 @@
+#
+# Copyright 2020 Autodesk
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+This file contains a set of reference helper functions which can be used to test and discover
+new capabilities provided by proxy accessor on a proxy shape.
+
+The end goal is to leverage UFE to provide integration of mixed data model compute with
+workflows like parenting (USD prim to Maya DAG object or vice versa), keying, or constraining.
+For the time being, this set of helper functions can be used as a reference to understand
+the concepts necessary to build tooling on top of the proxy accessor.
+"""
+
+from pxr import Sdf, Usd, UsdGeom, Vt
+from mayaUsd import lib as mayaUsdLib
+import maya.cmds as cmds
+import maya.OpenMaya as om
+import ufe
+
+def GetUfeSelection():
+    try:
+        return next(iter(ufe.GlobalSelection.get()))
+    except:
+        return None
+
+def GetDagAndPrimFromUfe(ufeObject):
+    if ufeObject == None:
+        return None, None
+        
+    segmentCount = len(ufeObject.path().segments)
+    if segmentCount == 0:
+        return None, None
+        
+    dagPath = str(ufeObject.path().segments[0])[6:]
```

Comment:
The fact that we have to deal with `|world` is something I really don't like. I tried your suggestion but it didn't work as a replacement for the existing code.

Reply:
I agree, I really don't like it either, and that's why I made the suggestion.  I wonder what didn't work; do you have a recent Maya?  "|world" should not be produced by ufe.PathString.string().  Otherwise we can have a look at this when you're back at work on Monday, but rest assured that we will get rid of [6:]...